### DISCUSSION TITLE
ReactPHP integration for Http2/3 instead relying on nginx(like)

### DISCUSSION THREAD

**Discussion by arunnabraham** [reactphp/reactphp#478] [General]
I wanted to know based on the discussion
https://github.com/reactphp/reactphp/discussions/470#discussion-4385990

is Http2/3 could be integrated with ReactPHP based on Nghttp2 https://github.com/nghttp2/nghttp2
we can also see implementations like QUIC here 

https://github.com/ngtcp2/ngtcp2
https://github.com/ngtcp2/nghttp3

is it possible to implement with FFI or is it easy as an extension? what are your thoughts?

**Comment by WyriHaximus** [comment]
We have plans to add HTTP 2 and 3 on our roadmap. One thing we probably have to consider is how we deal with different versions of the protocol, and that could cover having pluggable replacements for each version with an extension or FFI when possible.
